proto/viam/app/build/v1/build.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package viam.app.build.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/app/build/v1";
+
+// Note on ID types:
+// build_id: The 'parent' ID for a set of builds of module@version. Builds contain multiple jobs. This ID is globally unique.
+
+// 2023-11-28: Unstable API. Will change.
+service BuildService {
+  // Asynchronously start a build of a Viam module, with one job per platform.
+  rpc StartBuild(StartBuildRequest) returns (StartBuildResponse);
+  // Fetch logs from a single job of a single build.
+  rpc GetLogs(GetLogsRequest) returns (stream GetLogsResponse);
+  // List the jobs for a module ordered by (build start time, alphabetical platform).
+  rpc ListJobs(ListJobsRequest) returns (ListJobsResponse);
+}
+
+message StartBuildRequest {
+  // repo to build
+  string repo = 1;
+  // optional git ref; defaults to 'main'
+  optional string ref = 2;
+  // list of platforms to build
+  repeated string platform = 3;
+  // module_id as prefix:name pair
+  string module_id = 4;
+}
+
+message StartBuildResponse {
+  string build_id = 1;
```

Comment:
Shouldn't this be `repeated string build_id`? Based on the protos below, it feels like there will be a `build_id` generated per `platform`.

Or is JobInfo supposed to have `repeated string platform`

Reply:
the model is 'one build contains multiple jobs'. each job represents one platform.

(and this maps idiomatically onto most of the vendors we've shopped)

Reply:
The current impl is that each "build" kicks off multiple jobs (one per platform)

Reply:
Makes sense now. And no plans for a job_id. Its the `build_id + platform` that makes a job unique?